src/effects/VST3/VST3Effect.cpp

```diff
@@ -253,110 +252,29 @@ int VST3Effect::ShowClientInterface(wxWindow& parent, wxDialog& dialog,
    return 0;
 }
 
-bool VST3Effect::InitializePlugin()
-{
-   return true;
-}
-   
 std::shared_ptr<EffectInstance> VST3Effect::MakeInstance() const
 {
    return std::make_shared<VST3Instance>(*this, *mModule, mEffectClassInfo.ID());
 }
 
-bool VST3Effect::IsGraphicalUI()
-{
-   return mPlugView != nullptr;
-}
-
 std::unique_ptr<EffectUIValidator> VST3Effect::PopulateUI(ShuttleGui& S,
    EffectInstance& instance, EffectSettingsAccess &access)
 {
-   using namespace Steinberg;
+   bool useGUI { true };
+   GetConfig(*this, PluginSettings::Shared, wxT("Options"),
+            wxT("UseGUI"),
+            useGUI,
+            useGUI);
 
+   const auto vst3instance = dynamic_cast<VST3Instance*>(&instance);
    mParent = S.GetParent();
 
-   auto& vst3instance = *dynamic_cast<VST3Instance*>(&instance);
-   auto& wrapper = vst3instance.GetWrapper();
-
-   auto parent = S.GetParent();
-   if(GetType() == EffectTypeGenerate)
-   {
-      auto vSizer = std::make_unique<wxBoxSizer>(wxVERTICAL);
-      auto controlsRoot = safenew wxWindow(parent, wxID_ANY);
-      if(!LoadVSTUI(*wrapper.mEditController, controlsRoot))
-         mPlainUI = VST3ParametersWindow::Setup(*controlsRoot, *wrapper.mEditController, *wrapper.mComponentHandler);
-      vSizer->Add(controlsRoot);
-
-      auto &extra = access.Get().extra;
-      mDuration = safenew NumericTextCtrl(
-            parent, wxID_ANY,
-            NumericConverter::TIME,
-            extra.GetDurationFormat(),
-            extra.GetDuration(),
-            wrapper.mSetup.sampleRate,
-            NumericTextCtrl::Options{}
-               .AutoPos(true)
-         );
-      mDuration->SetName( XO("Duration") );
-
-      auto hSizer = std::make_unique<wxBoxSizer>(wxHORIZONTAL);
-      hSizer->Add(safenew wxStaticText(parent, wxID_ANY, _("Duration:")));
-      hSizer->AddSpacer(5);
-      hSizer->Add(mDuration);
-      vSizer->AddSpacer(10);
-      vSizer->Add(hSizer.release());
-
-      parent->SetMinSize(vSizer->CalcMin());
-      parent->SetSizer(vSizer.release());
-   }
-   else if(!LoadVSTUI(*wrapper.mEditController, parent))
-   {
-      mPlainUI = VST3ParametersWindow::Setup(
-         *parent,
-         *wrapper.mEditController,
-         *wrapper.mComponentHandler
-      );
-   }
-   wrapper.BeginParameterEdit(access);
-
-   mCurrentDisplayInstance = std::dynamic_pointer_cast<VST3Instance>(vst3instance.shared_from_this());
-   
-   return std::make_unique<DefaultEffectUIValidator>(*this, access);
-}
-
-bool VST3Effect::ValidateUI(EffectSettings &settings)
-{
-   if (mDuration != nullptr)
-      settings.extra.SetDuration(mDuration->GetValue());
-
-   assert(mCurrentDisplayInstance);
-
-   mCurrentDisplayInstance->GetWrapper().StoreSettings(settings);
-   mCurrentDisplayInstance->GetWrapper().FlushSettings(settings);
```

Comment:
The sequence of those two calls is changed in the Validator now, and I think that is correct